.github/workflows/release.yaml

```diff
@@ -0,0 +1,71 @@
+# Prefect Operator release workflow.
+name: Release
+
+on:
+  release:
+    types: [published]
+  pull_request:
+    branches:
+      - main
+
+permissions: {}
+
+jobs:
+  build-and-upload-manifests:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: jdx/mise-action@v2
+      - name: Build manifests
+        run: |
+          (cd config/manager && kustomize edit set image controller=prefecthq/prefect-operator:${{ github.ref_name }})
+          kustomize build config/crd > prefect-crds.yaml
+          kustomize build config/default > prefect.yaml
+      - name: Upload release assets
+        if: github.event_name == 'release'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh release upload ${{github.event.release.tag_name}} prefect-crds.yaml prefect.yaml
+
+  build-and-push-docker-image:
+    runs-on: ubuntu-latest
+    # The GitHub environments are created by Terraform and map to Docker Hub repositories:
+    # - dev: https://hub.docker.com/r/prefecthq/prefect-operator-dev
+    # - prod: https://hub.docker.com/r/prefecthq/prefect-operator
+    # The environment will be 'prod' if the GitHub event is a release. Otherwise, it will be 'dev'.
+    environment: ${{ github.event_name == 'release' && 'prod' || 'dev' }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          # These credentials are managed in Terraform. Depending on the 'environment' value above,
+          # these will either be the credentials for 'dev' or 'prod'.
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Docker image metadata
+        id: metadata
+        uses: docker/metadata-action@v5
+        with:
+          images: prefecthq/prefect-operator${{ github.event_name == 'pull_request' && '-dev' }}
+          tags: |
+            type=ref,event=pr
+            type=semver,pattern={{version}},event=tag
```

Comment:
where is this {{version}} template var coming from?

Reply:
It's apparently built in magic from https://github.com/docker/metadata-action?tab=readme-ov-file#semver. We use this in the [Prefect Prometheus release setup](https://github.com/PrefectHQ/prometheus-prefect-exporter/blob/4fe99739ac77ddd356c65cb51bbf0cce07ab8cfa/.github/workflows/release-build-deploy.yaml#L40) as well.